tests/ignite/metrics/regression/test_canberra_metric.py

```diff
@@ -58,45 +59,48 @@ def test_compute():
     assert canberra.pairwise([v1, v2])[0][1] == pytest.approx(np_sum)
 
 
-def test_integration():
-    def _test(y_pred, y, batch_size):
-        def update_fn(engine, batch):
-            idx = (engine.state.iteration - 1) * batch_size
-            y_true_batch = np_y[idx : idx + batch_size]
-            y_pred_batch = np_y_pred[idx : idx + batch_size]
-            return torch.from_numpy(y_pred_batch), torch.from_numpy(y_true_batch)
-
-        engine = Engine(update_fn)
+@pytest.mark.parametrize("n_times", range(5))
+@pytest.mark.parametrize(
+    "test_cases",
+    [
+        (torch.rand(size=(100,)), torch.rand(size=(100,)), 10),
+        (torch.rand(size=(100, 1)), torch.rand(size=(100, 1)), 20),
+    ],
+)
+def test_integration(n_times, test_cases, available_device):
+    y_pred, y, batch_size = test_cases
+    assert y_pred.dtype == torch.float32
+    assert y.dtype == torch.float32
 
-        m = CanberraMetric()
-        m.attach(engine, "cm")
+    def update_fn(engine, batch):
+        idx = (engine.state.iteration - 1) * batch_size
+        y_true_batch = y[idx : idx + batch_size].to(dtype=torch.float32)
+        y_pred_batch = y_pred[idx : idx + batch_size].to(dtype=torch.float32)
+        return y_pred_batch, y_true_batch
 
-        np_y = y.numpy().ravel()
-        np_y_pred = y_pred.numpy().ravel()
+    engine = Engine(update_fn)
 
-        canberra = DistanceMetric.get_metric("canberra")
+    m = CanberraMetric(device=available_device)
+    print(f"m's dtype: {m._double_dtype}")
+    assert m._device == torch.device(available_device)
 
-        data = list(range(y_pred.shape[0] // batch_size))
-        cm = engine.run(data, max_epochs=1).metrics["cm"]
+    m.attach(engine, "cm")
+    print(f"m's dtype again: {m._double_dtype}")
 
-        assert canberra.pairwise([np_y_pred, np_y])[0][1] == pytest.approx(cm)
+    canberra = DistanceMetric.get_metric("canberra")
 
-    def get_test_cases():
-        test_cases = [
-            (torch.rand(size=(100,)), torch.rand(size=(100,)), 10),
-            (torch.rand(size=(100, 1)), torch.rand(size=(100, 1)), 20),
-        ]
-        return test_cases
+    data = list(range(y_pred.shape[0] // batch_size))
+    cm = engine.run(data, max_epochs=1).metrics["cm"]
 
-    for _ in range(5):
-        # check multiple random inputs as random exact occurencies are rare
-        test_cases = get_test_cases()
-        for y_pred, y, batch_size in test_cases:
-            _test(y_pred, y, batch_size)
+    X = y_pred.cpu().numpy().reshape(len(y_pred), -1)
+    Y = y.cpu().numpy().reshape(len(y), -1)
```

Comment:
Let's avoid variables in upper case, lint wont pass most probably.
Also, x and y for y_pred and y are not appropriate names.